defs/run_in_workspace.bzl

```diff
@@ -23,12 +23,19 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-BASE=$(pwd)
-cd $(dirname $(readlink {root_file}))
-"$BASE/{cmd}" $@
+if [[ -n "${{BUILD_WORKSPACE_DIRECTORY:-}}" ]]; then
+  # Running from inside bazel
+  cd "${{BUILD_WORKSPACE_DIRECTORY}}"
+else
+  # Running from bazel-bin
+  cd "$(git rev-parse --show-toplevel)"
```

Comment:
so the only downside to this is that it won't work in a tarball export or other similar copy that is not git a git clone.

`bazel-bin` is also typically a symlink to a totally different location, so i'm not sure this would always work.

[thinking about alternatives...]

Reply:
I think you are fighting the wrong battle.

Right now it doesn't work unless called from inside bazel run. Now there are many scenarios where it will.

How is this not an improvement?

Reply:
i'm not quite convinced this actually does what we want unless you happen to already be in workspace root or a subdir of it?

Reply:
🤷‍♂

Reply:
I agree, but right now it also does not work when you're in the workspace root and/or a subdir of it. With this PR it will.

Getting `bazel build //foo:bar && bazel-bin/foo/bar` or `bazel build //:bar && ../bazel-bin/foo/bar` to work seems infinitely more important right now that whatever scenario you're concerned about. How are you expecting them to build `//foo:bar`?